### PR TITLE
Check if the game is end 

### PR DESCRIPTION
**Pull request by minaccchiang**
Add bag and winning score in Game state (setter and getter)
Add winning page in presenter

Check if the player wins the game or the bag is run out of tiles before changing to the next turn

### CODE REVIEW

**Review by ethanhonest — DISMISSED** [review]
The changes look good overall. I would say that you don't need a direct access point to Bag, since you have access to it through the Players (who share the same bag), but I think it's more convenient in the way that you do it. Good job implementing on such short notice, but please include some tests if possible for this use case.

**Review by petesaiacct — APPROVED** [review]
The code is clear, see no issues.

**Review by ethanhonest — APPROVED** [review]
New test looks like it follows the conventions, and my previous review of the other parts of the code still holds. Looks good to me

File: src/test/java/take_turn_use_case/TakeTurnTest.java [hunk 1]
```diff
@@ -1,5 +1,115 @@
...
+import java.util.*;
+
+import CoreEntities.Player.HumanPlayer;
+import CoreEntities.Player.Player;
+import Take_Turn.*;
+import core_entities.game_parts.*;
+import default_reference_values.DefaultBoardDataAccessObject;
+import io.ui.logic.Presenter;
+import launch_new_game_use_case.LaunchGameDataAccessObject;
+import launch_new_game_use_case.LaunchGameRequestModel;
```
Comment (petesaiacct) [inline src/test/java/take_turn_use_case/TakeTurnTest.java:13]: This is a tremendous amount of imports for a test. Lets revisit afterwards.

**Review by petesaiacct — APPROVED** [review]
The test, is a lot of imports. But the rest is great.

**Review by JosephUofT — APPROVED** [review]
Looks good. But we should look at the tests as Pete mentioned.

**Pull request merged**